staging/src/k8s.io/csi-translation-lib/translate.go

```diff
@@ -35,9 +35,36 @@ var (
 	}
 )
 
+// CSITranslator translates in-tree storage API objects to their equivalent CSI
+// API objects. It also provides many helper functions to determine whether
+// translation logic exists and the mappings between "in-tree plugin <-> csi driver"
+type CSITranslator interface {
+	TranslateInTreeStorageClassToCSI(inTreePluginName string, sc *storage.StorageClass) (*storage.StorageClass, error)
+	TranslateInTreeInlineVolumeToCSI(volume *v1.Volume) (*v1.PersistentVolume, error)
+	TranslateInTreePVToCSI(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+	TranslateCSIPVToInTree(pv *v1.PersistentVolume) (*v1.PersistentVolume, error)
+	IsMigratableIntreePluginByName(inTreePluginName string) bool
+	IsMigratedCSIDriverByName(csiPluginName string) bool
+	GetInTreePluginNameFromSpec(pv *v1.PersistentVolume, vol *v1.Volume) (string, error)
+	GetCSINameFromInTreeName(pluginName string) (string, error)
+	GetInTreeNameFromCSIName(pluginName string) (string, error)
+	IsPVMigratable(pv *v1.PersistentVolume) bool
+	IsInlineMigratable(vol *v1.Volume) bool
+}
+
+type csiTranslator struct{}
```

Comment:
I suggest that you just export this struct, and define the above interface in the packages that depend on these methods. This is the pattern recommended by go code review comments:

https://github.com/golang/go/wiki/CodeReviewComments#interfaces

That solves the testing problem, and the interface and test implementations in package need to define the methods that they need.

Reply:
Adding a method to this struct breaks all pure implementors, even if clients weren't using those methods.

Reply:
ah I had that before and was told to change it:
https://github.com/kubernetes/kubernetes/pull/82683#discussion_r324562465
https://github.com/kubernetes/kubernetes/pull/82683#discussion_r324882506

I could change it back if @misterikkit @jsafrane will agree. I prefer exporting the struct and I believe it's the idiomatic golang method

Reply:
I agree

Reply:
I agree. I was not aware of this recommendation. Sorry about the noise.